base/applications/cmdutils/whoami/whoami.c

```diff
@@ -80,35 +80,43 @@ BOOL WhoamiFree(VOID* Buffer)
 
 VOID* WhoamiGetTokenInfo(TOKEN_INFORMATION_CLASS TokenType)
 {
-    HANDLE hToken = 0;
-    DWORD dwLength = 0;
-    VOID* pTokenInfo = 0;
+    HANDLE hToken;
+    DWORD dwLength;
+    VOID* pTokenInfo = NULL;
+    BOOL bRet;
 
     if (OpenProcessToken(GetCurrentProcess(), TOKEN_READ, &hToken))
     {
-        GetTokenInformation(hToken,
-                            TokenType,
-                            NULL,
-                            dwLength,
-                            &dwLength);
+        bRet = GetTokenInformation(hToken,
+                                   TokenType,
+                                   NULL,
+                                   0,
+                                   &dwLength);
+        if (bRet || GetLastError() != ERROR_INSUFFICIENT_BUFFER)
```

Comment:
Did you ever test this code?

Reply:
Tested on LiveCD: results look unchanged.
What do you think is wrong?